src/server/router/application.ts

```diff
@@ -225,4 +231,196 @@ export const applicationRouter = router({
+          lastName: responsePreprocessing.get("irlsiZFKVJKD")?.text ?? "N/A",
+          phoneNumber:
+            responsePreprocessing.get("ceNTt9oUhO6Q")?.phone_number ?? "N/A",
+          email: responsePreprocessing.get("onIT7bTImlRj")?.email ?? "N/A",
+        },
+        mlhAgreement:
+          responsePreprocessing.get("F3vbQhObxXFa")?.boolean ?? false,
+        mlhCoc: responsePreprocessing.get("f3ELfiV5gVSs")?.boolean ?? false,
+      };
+    })[0];
+
+    if (converted === undefined) {
+      return {};
+    }
+
+    const pt = applicationSchema.partial();
+
+    type AutofillType = z.infer<typeof pt>;
+
+    const autofill: AutofillType = {};
+
+    if (converted.firstName !== "N/A") {
+      autofill["firstName"] = converted.firstName;
+    }
+    if (converted.lastName !== "N/A") {
+      autofill["lastName"] = converted.lastName;
+    }
+    if (converted.birthday !== undefined) {
+      autofill["birthday"] = converted.birthday.toISOString().slice(0, 10);
+    }
+
+    if (converted.major !== "N/A") {
+      autofill["studyMajor"] = converted.major;
+    }
+    if (converted.school !== "N/A") {
+      autofill["studyLocation"] = converted.school;
+    }
+    if (converted.willBeEnrolled !== false) {
+      autofill["studyEnrolledPostSecondary"] = converted.willBeEnrolled;
+    }
+    if (converted.graduationYear !== undefined) {
+      autofill["studyExpectedGraduation"] = converted.graduationYear
+        .toISOString()
+        .slice(0, 10);
+    }
+    if (converted.degree !== "N/A") {
+      autofill["studyDegree"] = converted.degree;
+    }
+
+    if (converted.hackathonCount !== "N/A") {
+      autofill["previousHackathonsCount"] = parseInt(converted.hackathonCount);
+    }
+    // emergencyContact
+
+    if (converted.emergencyContactInfo.firstName !== "N/A") {
+      autofill["emergencyContactName"] =
+        converted.emergencyContactInfo.firstName;
+    }
+
+    if (converted.emergencyContactInfo.lastName !== "N/A") {
+      autofill["emergencyContactName"] =
+        autofill["emergencyContactName"] +
+        " " +
+        converted.emergencyContactInfo.lastName;
+    }
+
+    if (converted.emergencyContactInfo.phoneNumber !== "N/A") {
+      autofill["emergencyContactPhone"] =
+        converted.emergencyContactInfo.phoneNumber;
+    }
+
+    if (converted.socialLinks !== "N/A") {
+      autofill["socialText"] = converted.socialLinks;
+    }
+
+    if (converted.tshirtSize !== "N/A") {
+      if (z.enum(["XS", "S", "M", "L", "XL"]).parse(converted.tshirtSize)) {
+        autofill["tshirtSize"] = converted.tshirtSize as
+          | "XS"
+          | "S"
+          | "M"
+          | "L"
+          | "XL";
+      }
+    }
+    // console.log(autofill)
+
+    return autofill;
+  }),
+  submitDh10: protectedProcedure
+    .input(applicationSchema)
+    .mutation(async ({ ctx, input }) => {
+      // make sure there is no existing application
+      const existing = await ctx.prisma.dH10Application.findFirst({
```

Comment:
this can lead to a race condition. just use `create` and try/catch, no?

Reply:
or use the transaction syntax coderabbit suggested

Reply:
should be fixed